docs/examples/talk-to-a-video.md

```diff
@@ -0,0 +1,83 @@
+We can use Google Gemini's [native video input](https://ai.google.dev/gemini-api/docs/vision?lang=python#prompting-video) capabilities to ask questions about [a video](https://www.youtube.com/watch?v=XXuIBHO4qa8).
+In this example, we upload a video file using Gemini's file API, and then pass the result using the [GenericArtifact](../reference/griptape/artifacts/generic_artifact.md) to the Agent.
+Note that because we are using Gemini-specific features, this will not work with other [Prompt Drivers](../griptape-framework/drivers/prompt-drivers.md).
+
+```python
+import time
+from griptape.structures import Agent
+from griptape.tasks import PromptTask
+from griptape.artifacts import ListArtifact, GenericArtifact, TextArtifact
+from griptape.config import GoogleStructureConfig
+import google.generativeai as genai
+
+video_file = genai.upload_file(path="tests/resources/griptape-comfyui.mp4")
+while video_file.state.name == "PROCESSING":
+    time.sleep(2)
+    video_file = genai.get_file(video_file.name)
+
+if video_file.state.name == "FAILED":
+    raise ValueError(video_file.state.name)
+
+agent = Agent(
+    config=GoogleStructureConfig(),
+    tasks=[
+        PromptTask(
+            lambda task: ListArtifact(
+                [
+                    GenericArtifact(video_file),
+                    TextArtifact(task.full_context["args"][0]),
```

Comment:
I'm not sure if this is generalizable, so happy to discuss. Hoping we can simplify how this is called so it's a bit more intuitive and consistent with how other PromptTasks work. It's a bit confusing that I'm not passing in a "prompt" anymore, but instead this blob of info. Also remembering to access `task.full_context` doesn't really follow our "easy to use" interface.

Something like the following:
```
PromptTask(prompt, other_artifacts=list[Artifact(video_file)])
```

Reply:
@emjay07 take a look at the updated example, I simplified the syntax quite a bit.

Reply:
Cool, I like it. 1:1 essentially with the API call right now.